sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/CompatibilityMatrix.java

```diff
@@ -130,6 +130,10 @@ public boolean supports(String sqVersion) {
       return compatibleVersionBySqVersion.containsKey(sqVersion);
     }
 
+    public boolean isArchived(String sqVersion){
+      return compatibleVersionBySqVersion.get(sqVersion).isEmpty();
+    }
```

Comment:
This checks is a bit difficult to read: we rely on the fact that the map contains an empty string for a given version to deduct that it is archived.